### PR TITLE
fix: all_ok check for empyt GRangesList (see issue #126)

### PR DESCRIPTION
**Pull request by lgatto**
This PR fixes
- the `all_ok` check documented in issue #126 
- the corresponding message that now thrown before filtering those that are ok and takes NAs into account.

### CODE REVIEW

**Comment by lgatto** [comment]
Out of curiosity, why do you return `NA` and not `FALSE` in that `all_ok` test? The latter wouldn't require to specify `na.rm = TRUE ` in the `sum()` in the message.

**Comment by lgatto** [comment]
PR updated to also handle empty elements later in the code. All details in #126

File: R/proteinToX.R [hunk 1]
```diff
@@ -324,18 +324,18 @@ proteinToGenome <- function(x, db, id = "name", idType = "protein_id") {
...
+    message(sum(are_ok, na.rm = TRUE), "/", length(are_ok), " OK")
     are_ok <- are_ok[!is.na(are_ok)]
     ## We've got now a list of GRanges
-    message(sum(are_ok), "/", length(are_ok), " OK")
     ## Perform the mapping for each input range with each mapped cds
     res <- mapply(
         cds_genome, as(coords_cds, "IRangesList"), as(x, "IRangesList"),
         FUN = function(gnm, cds, prt) {
-            if (is.null(gnm)) {
+            if (is.null(gnm) | !length(gnm)) {
```
Comment (jorainer) [inline R/proteinToX.R]: It should suffice to just have `if (!length(gnm))` here, since `lenght(NULL)` is also `0`.

Comment (jorainer) [reply R/proteinToX.R]: In addition, can you please change

```r
mcols(maps)$protein_start <- start(prt)
mcols(maps)$protein_end <- end(prt)
```

below to 

```r
mcols(maps) <- cbind(mcols(maps), 
                                     protein_start = start(prt),
                                     protein_end = end(prt))
```

`$` on a `DataFrame` is for some reasons slower than `cbind`.

**Review by jorainer — CHANGES_REQUESTED** [review]
Thanks @lgatto ! just some minor change requests.

File: R/proteinToX.R (hunk 1 above)
Comment (lgatto) [reply R/proteinToX.R]: All addressed in PR update - well spotted that `length(gnm)` was enough!

**Review by jorainer — APPROVED** [review]
Thanks @lgatto ! I will merge and check locally.

**Comment by lgatto** [comment]
That's a good start for 2022! :-) 

All the best to you and your family!

**Pull request merged**